### DISCUSSION TITLE
How to install a nested package.json in a Workspaces project?

### DISCUSSION THREAD

**Discussion by corysimmons** [pnpm/pnpm#7256] [Q&A]
```
- packages
	- workspace-a
		- package.json // works fine
	- workspace-b
		- package.json // works fine
		- x
			- y
				- z
					- package.json // how can I install this?
- pnpm-workspace.yaml
- package.json
```

Even if I manually navigate to `packages/workspace-b/x/y/z` then run `pnpm i`, it just installs the Workspaces' package.jsons.

This works if I cd to that `z` dir and run `npm i` or `yarn`.

**Comment by woifes** [comment]
How does your ```pnpm-workspace.yaml``` look like?
If I do it like:
```yaml
packages:
  - 'workspace-a'
  - 'workspace-b'
  - 'workspace-b/**'
```
It works

**Reply by corysimmons** [reply]
It was just

```yaml
packages:
  - 'packages/**'
```

But it doesn't seem to automatically install the deps in nested directories.

**Reply by johnformio** [reply]
Did you find a solution to this?

**Reply by cocode32** [reply]
Personally, I just ran `pn i` in the package folder. After setting `packages:` in the `yml` file, my `pn i` actually worked, but I was `cd`'d into that directory, because I need to run `pn dev` from that directory. Don't know if it works from the root, but if you set the projects in the yaml, and then just go into your directory, you'll have access to it.

**Comment by ryux1** [comment]
`pnpm` is finding the parent workspace, so changing into `z` does not make it a standalone project by itself.

If `z` should be a workspace package, include it in `pnpm-workspace.yaml` (either explicitly or with a matching `**` glob):

```yaml
packages:
  - 'packages/*'
  - 'packages/workspace-b/x/y/z'
```

Then `pnpm install` at the workspace root will include it. You can target only that package with `pnpm --filter nested-z install`, where `nested-z` is the `name` in its `package.json`.

If `z` is intentionally independent from the parent workspace, use:

```sh
cd packages/workspace-b/x/y/z
pnpm install --ignore-workspace
```

`--ignore-workspace` tells pnpm to ignore the parent `pnpm-workspace.yaml` and perform a normal standalone install. (`--ignore-workspace-root-check` is a different option.) The workspace package globs are documented here: https://pnpm.io/settings#packages

---
Written by an agent (OpenAI Codex, GPT-5).

**Reply by corysimmons** [reply]
> <img width="288" height="45" alt="image" src="https://github.com/user-attachments/assets/3aec68f5-9b37-4176-9091-07a61bd4b1c7" />

<img width="200" height="220" alt="image" src="https://github.com/user-attachments/assets/ef527a10-f632-4e13-a19b-68941093c90f" />

**Reply by sxflynn** [reply]
This is the world we live in now. At least, they disclosed their use of AI.

**Reply by ryux1** [reply]
Give it time—it’ll be writing both the answers and the drive-by comments soon.
